cheatsheets/Database_Security_Cheat_Sheet.md

```diff
@@ -0,0 +1,134 @@
+# Database Security Cheat Sheet
+
+## Introduction
+
+This cheat sheet provides guidance on securely configuring and using the SQL and NoSQL databases. It is intended to be used by application developers when they are responsible for managing the databases, in the absence of a dedicated database administrator (DBA). For details about protecting against SQL Injection attacks, see the [SQL Injection Prevention Cheat Sheet](SQL_Injection_Prevention_Cheat_Sheet.md).
+
+## Contents
+
+- [Connecting to the Database](#connecting-to-the-database)
+  - [Transport Layer Protection](#transport-layer-protection)
+- [Authentication](#authentication)
+  - [Storing Database Credentials](#storing-database-credentials)
+- [Permissions](#permissions)
+- [Database Configuration and Hardening](#database-configuration-and-hardening)
+  - [Microsoft SQL Server](#microsoft-sql-server)
+  - [MySQL and MariaDB](#mysql-and-mariadb)
+  - [PostgreSQL](#postgresql)
+  - [MongoDB](#mongodb)
+  - [Redis](#redis)
+
+## Connecting to the Database
+
+The backend database used by the application should be isolated as much as possible, in order to prevent malicious or undesirable users from being able to connect to it. Exactly how this is achieved will depend on the system and network architecture. The following options could be used to protect it:
+
+- Disabling network (TCP) access and requiring all access is over a local socket file or named pipe.
+- Configuring the database to only bind on localhost.
+- Restricting access to the network port to specific hosts with firewall rules.
+- Placing the database server in a separate DMZ isolated from the application server.
+
+Similar protection should be implemented to protect any web-based management tools used with the database, such as phpMyAdmin.
+
+When an application is running on an untrusted system (such as a thick-client), it should always connect to the backend through an API that can enforce appropriate access control and restrictions. Direct connections should **never** be made from a thick client to the backend database.
```

Comment:
access controls, plural

Reply:
"Access control" is valid, it's referring to the concept, rather than individual controls.